processor/src/chiplets/hasher/trace.rs

```diff
@@ -104,25 +104,19 @@ impl HasherTrace {
     /// Copies section of trace from the given range of start and end rows at the end of the trace.
     /// The hasher state of the last row is copied to the provided state input.
     pub fn copy_trace(&mut self, state: &mut [Felt; STATE_WIDTH], range: Range<usize>) {
-        let mut hasher_state: [Felt; STATE_WIDTH] = Default::default();
-        let mut selectors: [Felt; NUM_SELECTORS] = Default::default();
-
-        for row in range {
-            for (col, selector) in selectors.iter_mut().enumerate() {
-                *selector = self.selectors[col][row];
-            }
-
-            for (col, state) in hasher_state.iter_mut().enumerate() {
-                *state = self.hasher_state[col][row];
-            }
+        for selector in self.selectors.iter_mut() {
+            selector.extend_from_within(range.clone());
```

Comment:
`Range` don't implement `Copy` for historic reasons. The value is not boxed, so this should still be very efficient even if the compiler failed to optimize it.